doc/netplan.md

```diff
@@ -72,7 +72,7 @@ Physical devices
 
 Virtual devices
 
-:  (Examples: veth, bridge, bond) These are fully under the control of the
+:  (Examples: veth, bridge, bond, vrf) These are fully under the control of the
```

Comment:
question (blocking): Seeing `vrf` in this list, how is revertability handled in `netplan try`?

Does `networkd` handle it on its own?

Reply:
This is a very good question! And I needed to do some investigation in order to answer this. But this is working as expected (and similar to `vlans`): `vrfs` are cleaned up on `netplan apply` (if using the `--state` parameter properly) or on `netplan try` (using the `--config-file` parameter). `netplan try --state ...` still behaves a bit odd, but that's unrelated to this PR.

VRFs are different to bonds/bridges, in that the backend (networkd/NM) does not need to initialize certain parameters, which cannot be changed afterwards and is therefore much more similar to a VLAN interface, which can just be deleted.
So I think this is all fine!

```
$ cat /etc/netplan/main_test.yaml 
network:
  version: 2
  renderer: networkd
  ethernets:
    enp0s31f6:
      dhcp4: true
      dhcp6: true
$ cat ./test.yaml
network:
  vrfs:
    vrf0:
      table: 1004
      interfaces: [enp0s31f6]
$ sudo LD_LIBRARY_PATH=. PYTHONPATH=. src/netplan.script --config-file=test.yaml try
Do you want to keep these settings?


Press ENTER before the timeout to accept the new configuration


Changes will revert in 111 seconds

# === meanwhile in another terminal === 
$ ip l show vrf0
110: vrf0: <NOARP,MASTER,UP,LOWER_UP> mtu 65575 qdisc noqueue state UP mode DEFAULT group default qlen 1000
    link/ether a2:f7:52:d9:7c:5c brd ff:ff:ff:ff:ff:ff
# === terminal switch ===
^C
Reverting.
$ ip l show vrf0
Device "vrf0" does not exist.
```

(I actually found a bug/typo while testing this, which I fixed in b39dbea179f5b81609af002fd42986dccb97d1fc)